lib/danger/danger_core/dangerfile.rb

```diff
@@ -159,58 +160,60 @@ def parse(path, contents = nil)
+        ui.puts "Your #{path.basename} has had smart quotes sanitised. " \
+          'To avoid issues in the future, you should not use ' \
+          'TextEdit for editing it. If you are not using TextEdit, ' \
+          'you should turn off smart quotes in your editor of choice.'.red
       end
 
       if contents.include?("puts")
-        puts "You used `puts` in your Dangerfile. To print out text to GitHub use `message` instead"
+        ui.puts "You used `puts` in your Dangerfile. To print out text to GitHub use `message` instead"
       end
 
       self.defined_in_file = path
-      instance_eval do
-        # rubocop:disable Lint/RescueException
-        begin
-          # rubocop:disable Eval
-          eval(contents, nil, path.to_s)
-          # rubocop:enable Eval
-        rescue Exception => e
-          message = "Invalid `#{path.basename}` file: #{e.message}"
-          raise DSLError.new(message, path, e.backtrace, contents)
+      ui.section("Running:") do
+        instance_eval do
+          # rubocop:disable Lint/RescueException
+          begin
+            # rubocop:disable Eval
+            eval(contents, nil, path.to_s)
+            # rubocop:enable Eval
+          rescue Exception => e
+            message = "Invalid `#{path.basename}` file: #{e.message}"
+            raise DSLError.new(message, path, e.backtrace, contents)
+          end
+          # rubocop:enable Lint/RescueException
         end
-        # rubocop:enable Lint/RescueException
       end
     end
 
     def print_results
       status = status_report
       return if (status[:errors] + status[:warnings] + status[:messages] + status[:markdowns]).count == 0
 
-      puts ""
-      puts "danger results:"
-      [:errors, :warnings, :messages].each do |current|
-        params = {}
-        params[:rows] = status[current].map { |item| [item] }
-        next unless params[:rows].count > 0
-        params[:title] = case current
-                         when :errors
-                           current.to_s.capitalize.red
-                         when :warnings
-                           current.to_s.capitalize.yellow
-                         else
-                           current.to_s.capitalize
-                         end
-
-        puts ""
-        puts Terminal::Table.new(params)
-        puts ""
-      end
+      ui.section('Results:') do
+        [:errors, :warnings, :messages].each do |current|
+          params = {}
+          params[:rows] = status[current].map { |item| [item] }
+          next unless params[:rows].count > 0
+          params[:title] = case current
+                           when :errors
+                             current.to_s.capitalize.red
+                           when :warnings
+                             current.to_s.capitalize.yellow
+                           else
+                             current.to_s.capitalize
+                           end
+          ui.puts Terminal::Table.new(params)
```

Comment:
no, the puts are necessary -- they add blank lines before & after the table. all that was unnecessary was the argument